tests/test_environment_LocalDescriptors.py

```diff
@@ -110,6 +112,181 @@ def test_repr(self):
         comp = freud.environment.LocalDescriptors(4, 8, 0.5, True)
         self.assertEqual(str(comp), str(eval(repr(comp))))
 
+    def test_ql(self):
+        """Check if we can reproduce Steinhardt OPs."""
+        def get_Ql(p, descriptors, nlist):
+            """Given a set of points and a LocalDescriptors object (and the
+            underlying neighborlist, compute the per-particle Steinhardt order
+            parameter for all :math:`l` values up to the maximum quantum number
+            used in the computation of the descriptors."""
+            Qbar_lm = np.zeros((p.shape[0], descriptors.sph.shape[1]),
+                               dtype=np.complex128)
+            num_neighbors = descriptors.sph.shape[0]/p.shape[0]
+            for i in range(p.shape[0]):
+                indices = nlist.index_i == i
+                Qbar_lm[i, :] = np.sum(descriptors.sph[indices, :],
+                                       axis=0)/num_neighbors
+
+            Ql = np.zeros((Qbar_lm.shape[0], descriptors.l_max+1))
+            for i in range(Ql.shape[0]):
+                for l in range(Ql.shape[1]):
+                    for k in range(l**2, (l+1)**2):
+                        Ql[i, l] += np.absolute(Qbar_lm[i, k])**2
+                    Ql[i, l] = np.sqrt(4*np.pi/(2*l + 1) * Ql[i, l])
+
+            return Ql
+
+        # These exact parameter values aren't important; they won't necessarily
+        # give useful outputs for some of the structures, but that's fine since
+        # we just want to check that LocalDescriptors is consistent with
+        # Steinhardt.
+        num_neighbors = 6
+        l_max = 12
+        r_max = 2
+
+        for struct_func in [make_sc, make_bcc, make_fcc]:
+            box, points = struct_func(5, 5, 5)
+
+            # In order to be able to access information on which particles are
+            # bonded to which ones, we precompute the neighborlist
+            nn = freud.locality.NearestNeighbors(r_max, num_neighbors)
```

Comment:
Let's use a NeighborQuery API instead, so that we won't have to refactor this later.

Reply:
Good call, I was thinking that we shouldn't use the new API on master but should be fine.